FTBFS on Debian with Boost 1.61

I have refreshed the packaging for `v3.1.1` and attempted to build on a Debian unstable build machine. I got a Boost error (Boost 1.61), which I don't get on my local machine running testing (Boost 1.58). Another difference is the compiler: gcc 6.1.1 on unstable, gcc 5.4.0 on testing.

The complete build log is available [here](http://debomatic-amd64.debian.net/distribution#unstable/shark/3.1.1+ds1-1/buildlog). The relevant portions are:

cmake output:
```
cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DBUILD_SHARED_LIBS=ON -DENABLE_CBLAS=ON -DENABLE_OPENMP=ON -DSHARK_INSTALL_DOC_DIR=share/doc/shark -DSHARK_INSTALL_EXAMPLE_DIR=share/doc/shark/examples -DBUILD_TESTING=ON -DBUILD_DOCUMENTATION=OFF -DBUILD_EXAMPLES=OFF
-- The C compiler identification is GNU 6.1.1
-- The CXX compiler identification is GNU 6.1.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Will build: None
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.61.0
-- Found the following Boost libraries:
--   system
--   date_time
--   filesystem
--   program_options
--   serialization
--   thread
--   unit_test_framework
--   chrono
--   atomic
-- Using boost from 
-- OpenMP found
-- Checking HDF5 installation: HDF5 installation seems ok.
-- CBLAS FOUND: GENERIC with include directory /usr/include
-- Configuring done
-- Generating done
-- Build files have been written to: /<<BUILDDIR>>/shark-3.1.1+ds1/obj-x86_64-linux-gnu
```

build error:
```
/<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Core/utility/ZipPair.h:145:28: error: 'make_iterator_range' was not declared in this scope
  return make_iterator_range(iterator(begin1,begin2),iterator(end1,end2));
         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Core/utility/ZipPair.h:145:28: note: suggested alternative:
In file included from /usr/include/boost/range/iterator_range.hpp:13:0,
                 from /usr/include/boost/range.hpp:20,
                 from /<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Core/utility/Range.h:35,
                 from /<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Core/utility/Iterators.h:39,
                 from /<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Core/utility/functional.h:39,
                 from /<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Models/Trees/BinaryTree.h:40,
                 from /<<BUILDDIR>>/shark-3.1.1+ds1/include/shark/Models/Trees/LCTree.h:39,
                 from /<<BUILDDIR>>/shark-3.1.1+ds1/obj-x86_64-linux-gnu/examples/Unsupervised/HierarchicalClustering.cpp:35:
/usr/include/boost/range/iterator_range_core.hpp:853:9: note:   'boost::make_iterator_range'
         make_iterator_range( const Range& r,
         ^~~~~~~~~~~~~~~~~~~
examples/CMakeFiles/HierarchicalClustering.dir/build.make:69: recipe for target 'examples/CMakeFiles/HierarchicalClustering.dir/Unsupervised/HierarchicalClustering.cpp.o' failed
```

--- a/Test/LinAlg/BLAS/triangular_matrix.cpp
+++ b/Test/LinAlg/BLAS/triangular_matrix.cpp
@@ -5,6 +5,8 @@
 #include <shark/LinAlg/BLAS/triangular_matrix.hpp>
 #include <shark/LinAlg/BLAS/matrix.hpp>
 
+#include <iostream>
+
 using namespace shark;
 using namespace blas;
 

--- a/include/shark/Core/utility/ZipPair.h
+++ b/include/shark/Core/utility/ZipPair.h
@@ -142,7 +142,7 @@ boost::iterator_range<
 >
 zipPairRange(Iterator1 begin1, Iterator1 end1, Iterator2 begin2,Iterator2 end2){
 	typedef PairIterator<PairType,Iterator1,Iterator2> iterator;
-	return make_iterator_range(iterator(begin1,begin2),iterator(end1,end2));
+	return boost::make_iterator_range(iterator(begin1,begin2),iterator(end1,end2));
 }
 
 template<class PairType, class Range1, class Range2>

